lang/cs/Org.Apache.REEF.Wake/Remote/IWritable.cs

```diff
@@ -0,0 +1,44 @@
+﻿using System.IO;
+using System.Threading;
+using System.Threading.Tasks;
+
+namespace Org.Apache.REEF.Wake.Remote
+{
+    /// <summary>
+    /// Interface that classes should implement if they need to be readable to and writable 
+    /// from the stream. It is assumed that the classes inheriting this interface will have a 
+    /// default empty constructor
+    /// </summary>
+    public interface IWritable
+    {
+        /// <summary>
+        /// Read the class fields from the stream.
+        /// </summary>
+        /// <param name="stream">The stream from which to read</param>
+        /// <param name="optionalParameters">The optional parameters to be passed to the reader.
```

Comment:
Can you give an example? The Hadoop `Writable` which this is based on didn't need the additional parameters. I am concerned because they add type-unsafety to the code.